lib/src/local_working_copy.rs

```diff
@@ -1177,24 +1176,16 @@ impl TreeState {
         Ok(FileState::for_file(executable, size, &metadata))
     }
 
-    #[cfg_attr(windows, allow(unused_variables))]
     fn write_symlink(&self, disk_path: &Path, target: String) -> Result<FileState, CheckoutError> {
-        #[cfg(windows)]
-        {
-            println!("ignoring symlink at {}", disk_path.display());
-        }
-        #[cfg(unix)]
-        {
-            let target = PathBuf::from(&target);
-            symlink(&target, disk_path).map_err(|err| CheckoutError::Other {
-                message: format!(
-                    "Failed to create symlink from {} to {}",
-                    disk_path.display(),
-                    target.display()
-                ),
-                err: err.into(),
-            })?;
-        }
+        let target = PathBuf::from(&target);
+        try_symlink(&target, disk_path).map_err(|err| CheckoutError::Other {
+            message: format!(
+                "Failed to create symlink from {} to {}",
+                disk_path.display(),
+                target.display()
+            ),
+            err: err.into(),
+        })?;
```

Comment:
Shouldn't we continue to just print a warning if developer mode is not enabled? Otherwise this commit gives the users who don't have developer mode enabled a worse experience, IIUC.

Reply:
I implemented this, but found out that skipping the symlink doesn't actually make the checkout work - it's too robust :) The status quo is that this fails entirely:
```
ignoring symlink at \\?\C:\Users\banana\Documents\code\jj\docs\cli-reference.md
Internal error: Failed to check out commit ba70a0c722af06485cb47bf5609eddfc60290a48: Failed to stat file \\?\C:\Users\banana\Documents\code\jj\docs\cli-reference.md: The system cannot find the file specified. (os error 2)
```

I've added a case to CheckoutError so that we get a single structured error message, rather than a skip followed by a generic internal error. To make checkout *work* with symlink support disabled, we'd have to do something like Git's solution of encoding them as text files (and even that doesn't give you an actually-functional source tree).